app.go

```diff
@@ -550,7 +551,7 @@ func (a *Application) DeliverTx(txBytes []byte) abci.ResponseDeliverTx {
 	r, err := a.processTx(txBytes, false)
 	if err != nil {
 		log.Error(fmt.Sprintf("DeliverTx: %s", err.Error()))
-		return abci.ResponseDeliverTx{Code: 1, Log: err.Error()}
+		return abci.ResponseDeliverTx{Code: 1, Log: err.Error(), Data: ttypes.Tx(txBytes).Hash()}
```

Comment:
This is a consensus/replay breaking change, the `Data` field is part of the block hash, so you can't change it in `DeliverTx` without a feature flag, though in theory you should be able to change it in `CheckTx` without breaking anything.

Reply:
Removed. It's possible to calculate the tx hash in advance with `"github.com/tendermint/tendermint/types".Tx(tx-input-Bytes).Hash()`.